resources/notes.py

```diff
@@ -26,4 +26,14 @@ class Note(Resource):
     def delete(self, ticket_id, id):
         ticket = TicketModel.find(ticket_id)
         ticket.notes.delete(NotesModel.find(id))
+
         return {"message": "Note deleted"}
+
+    @pm_level_required
+    def patch(self, ticket_id, id):
+        ticket = TicketModel.find(ticket_id)
+        updated_note = NotesModel.update(
+            schema=NotesSchema, id=id, payload={"text": request.json["text"]}
+        )
+        ticket.notes.update(NotesModel.find(id), updated_note)
+        return updated_note.json()
```

Comment:
Let me know your thoughts on this. I think the flow I would like to see is something like this:

```python
ticket = TicketModel.find(ticket_id)
note = ticket.notes.find(id)
return note.update(schema=NotesSchema, payload=request.json).json()
```
What this should do is automatically return a 404 if ticket or note is not found.
If note exists then we can update it and serialize the new note. And then of course Marshmallow will handle any validation errors for us automatically.

What do you think?

Reply:
So add a find method to the nobiru list, and then use the note.update to actually change the note entry in the db?  I think that makes a lot of sense, and less clunky than my solution.  :).